### PR TITLE
cjdns: generate .conf file on install + add plist section

### PR DESCRIPTION
**Pull request by Sija**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/cjdns.rb [hunk 1]
```diff
@@ -20,6 +20,40 @@ def install
     (pkgshare/"test").install "build_darwin/test_testcjdroute_c" => "cjdroute_test"
   end
 
+  def post_install
+    `#{bin}/cjdroute --genconf > #{HOMEBREW_PREFIX}/etc/cjdroute.conf`
```
Comment (ilovezfs) [inline Formula/cjdns.rb]: This should be done in `install` and
```
(buildpath/"cjdroute.conf").write Utils.popen_read(bin/"cjdroute", "--genconf")
etc.install "cjdroute.conf"
```

**Review by ilovezfs — CHANGES_REQUESTED** [review]

File: Formula/cjdns.rb (hunk 1 above)
Comment (Sija) [reply Formula/cjdns.rb]: Unfortunately it doesn't work.

Comment (ilovezfs) [reply Formula/cjdns.rb]: They're equivalent. So you need to elaborate.

Comment (Sija) [reply Formula/cjdns.rb]: When using solution from your comment `cjdroute.conf` file is missing, I don't know what else to tell you.

Comment (ilovezfs) [reply Formula/cjdns.rb]:
```
🍺  /usr/local/Cellar/cjdns/20.1: 7 files, 1.7MB, built in 24 seconds
iMac-TMP:~ joe$ ls /usr/local/etc/cjdroute.conf 
/usr/local/etc/cjdroute.conf
```

**Comment by fxcoudert** [comment]
Thanks @Sija for the pull request!

File: Formula/cjdns.rb (hunk 1 above)
Comment (Sija) [reply Formula/cjdns.rb]:
```
🍺  /usr/local/Cellar/cjdns/20.1: 7 files, 1.7MB
~ ❯ ls /usr/local/etc/cjdroute.conf
ls: /usr/local/etc/cjdroute.conf: No such file or directory
```

Comment (Sija) [reply Formula/cjdns.rb]: If I'm not mistaken `install` step is skipped when formula is poured. That's the reason I've used `post_install` hook.

Comment (ilovezfs) [reply Formula/cjdns.rb]: Yes, you are mistaken.

Comment (Sija) [reply Formula/cjdns.rb]: So why `install` method is not invoked then (and hence missing `cjdroute.conf`)?

Comment (ilovezfs) [reply Formula/cjdns.rb]: The files that have been installed in etc while the bottle was built are collected in the bottles's hidden .bottle directory and installed from there even though the `install` method itself does not run when pouring.

Comment (Sija) [reply Formula/cjdns.rb]: So how can I generate config file, on the fly, if `install` method is **not** being run?

Comment (ilovezfs) [reply Formula/cjdns.rb]: You need to trust me on how this works. Unfortunately, I don't have time to elaborate beyond what I said above.

Comment (Sija) [reply Formula/cjdns.rb]: btw, you've said I'm mistaken, but in the very next message you proved me right by saying:

> [...] even though the install method itself **does not run when pouring**.

**Comment by ilovezfs** [comment]
Thanks for the PR but closing since we're not going to argue.

**Comment by Sija** [comment]
Thanks for being rude. I thought you care for your users, not your ego.

**Pull request closed without merging**